assets/src/stories-editor/store/reducer.js

```diff
@@ -19,10 +20,10 @@ import { isValidAnimationPredecessor } from './selectors';
  *
  * @return {Object} Updated state.
  */
-export function animations( state = {}, action ) {
-	const newAnimationOrder = { ...state };
+export function animations( state = {}, action ) { // eslint-disable-line complexity
```

Comment:
Cyclomatic complexity hits 22 for this function, which definitely hints that a rewrite or reorganisation would be a good idea - if for nothing else to break out the different cases into smaller functions. It's a 1000 sloc function!

Redux has [a good guide](https://redux.js.org/recipes/structuring-reducers/refactoring-reducer-example) for breaking reducers into smaller components.

Reply:
Extracted the individual cases into their own functions now. Not 100% DRY still, but it's a start.